cluster-api/src/test/java/io/aiven/klaw/clusterapi/services/SchemaServiceRegisterSchemaTest.java

```diff
@@ -412,12 +412,18 @@ public void givenSchemaWithForceRegisterEnabled_NotFoundGetGlobalCompatibility()
             any(HashMap.class)))
         .thenReturn(createErrorCompatibilityResponseEntity(HttpStatus.NOT_FOUND))
         .thenReturn(createErrorCompatibilityResponseEntity(HttpStatus.NOT_FOUND));
+    when(restTemplate.exchange(
+            eq(REGISTRY_URL),
+            any(HttpMethod.class),
+            any(HttpEntity.class),
+            eq(new ParameterizedTypeReference<Map<String, String>>() {}),
+            any(HashMap.class)))
+        .thenReturn(createErrorCompatibilityResponseEntity(HttpStatus.NOT_FOUND))
+        .thenReturn(createErrorCompatibilityResponseEntity(HttpStatus.NOT_FOUND));
     schemaService.registerSchema(schemaReq);
     // change schema compatibility to NONE, once
     verify(restTemplate, times(1)).put(any(), any(), eq(String.class));
 
-    verify(restTemplate, times(2))
-        .postForEntity(anyString(), any(HttpEntity.class), eq(String.class));
     // 2 calls to get the current SchemaCompatibility (subject, global) as isForce==false
```

Comment:
I think the anyString() is the problem if possible itd be great to put in an eq(whatever the string url is) to add that extra bit of verification

Reply:
Also is there any unit test for getting the schema version after?

Reply:
> I think the anyString() is the problem if possible itd be great to put in an eq(whatever the string url is) to add that extra bit of verification

resttemplate.exchange method is now added for schema versions, and it has eq(REGISTRY_URL).